pakyow-core/lib/pakyow/core/controller.rb

```diff
@@ -53,8 +55,8 @@ def process
       request.error = error
 
       catch :halt do
-        app.state_for(:router).each do |router|
-          if status = router.exception(error.class, context: self, handlers: handlers, exceptions: exceptions)
+        if failed_router
```

Comment:
Instead of checking, we could use the lonely operator:

```ruby
if status = current_router&.exception
```

We can get away with this because 1.0 will only support 2.4+ (this operator was introduced in 2.3).